syntax = "proto3";

// 外部用户信息，商业化应用的顾客
message Customer {
  // 客户注册渠道
  enum Method {
    // 微信
    WECHAT = 0;
    // 支付宝
    ALIPAY = 1;
  }
  // 顾客id(在IAM中)
  string id = 1;
  // 注册时间
  int64 create_time = 2;
  // 更新时间
  int64 update_time = 3;
  // 注销时间
  int64 delete_time = 4;
  // 手机号码
  string phone = 5;
  // 性别
  string sex = 6;
  // 城市
  string city = 7;
  // 省份
  string province = 8;
  // 国家
  string country = 9;
  // 头像地址
  string avatar = 10;
  // 别称
  string nickname = 11;
  // 注册方式/渠道
  Method method = 12;
  oneof profile {
    WechatProfile wechat_profile = 13;
    AlipayProfile alipay_profile = 14;
    // 非第三方授权需要密码
    string password = 15;
  }

}

// 用户在微信平台的信息
message WechatProfile {
  // 微信平台的openid
  string openid = 1;
  // 用户特权
  string privilege = 2;
  // 用户统一标识
  string union_id = 3;
}

// 用户在支付宝平台的信息
message AlipayProfile {
  // 支付宝平台的user_id
  string user_id = 1;
}
